### PR TITLE
Added Storybook

### LINKED ISSUE #42: [Storybook] Dodati storybook
- napraviti storybook da imamo pregled svih komponenti koje smo koristili u projektu, [link](https://storybook.js.org/)
- kako aleksdev kaže storybook ne radi dobro sa material UI al naš starter po defaultu valjda nema material UI pa možemo ubaciti storybook
- dokumentirati kako se dodaju nove komponente, šta sve treba napravit da se prikažu u storybooku, kako možemo vidjeti storybook itd.
- trenutno nemamo niti jednu komponentu u starteru pa nemamo šta ubacit u storybook. Možda napraviti još view sa favorite pokemonima, taj nek prikazuje listu pokemona koje smo odabrali u favoritese (nek se sprema u localstorage) - onda cemo listu pokemona imat ko komponentu za storbook. Razmisliti još ima li smisla i ove ostale stvari koje su nam sad div-ovi izbacit u komponente samo da pokažemo na njima kako radimo storybook i to sve

### PR DESCRIPTION
**Pull request by DorijanH**
Closes #42

### CODE REVIEW

**Comment by AleksandarDev** [comment]
Is this working when you change theme in theme.js?

**Comment by DorijanH** [comment]
Yes. Yes it is

**Comment by AleksandarDev** [comment]
Component `UserInformation` is missing story.

**Comment by AleksandarDev** [comment]
Button primary color not applied

image

**Review by AleksandarDev — APPROVED** [review]

**Comment by AleksandarDev** [comment]
@DorijanH @radovix requested:
>Pagination can be one component. @DorijanH just add some more logic to it (e.g. showing numbers of two previous and two next pages so that user can skip some pages).

Did you discussed this with him to skip for now?

**Comment by DorijanH** [comment]
@AleksandarDev Not really. This was suggested only to fill the Storybook with something because the initial premise was that we don't have anything to document in Storybook. In the meantime, I’ve set up stories for simpler pages that don’t depend on view models and the two components we use in the starter. So I thought that we don't need to dissect the rest of the application into components just to document something

**Comment by AleksandarDev** [comment]
> @AleksandarDev Not really. This was suggested only to fill the Storybook with something because the initial premise was that we don't have anything to document in Storybook. In the meantime, I’ve set up stories for simpler pages that don’t depend on view models and the two components we use in the starter. So I thought that we don't need to dissect the rest of the application into components just to document something

Internally we concluded that views shouldn't be in storybook, except eg. 500 page, and 404. Only shared components from `components` folder are documented in storybook.

File: .eslintrc [hunk 1]
```diff
@@ -3,7 +3,8 @@
   "extends": [
 		"airbnb",
 		"plugin:react/recommended",
-		"plugin:@next/next/recommended"
+		"plugin:@next/next/recommended",
+		"plugin:storybook/recommended"
```
Comment (radovix) [inline .eslintrc:7]: I think the order here matters so I would put `"plugin:storybook/recommended"` right after `"airbnb"`.

Comment (DorijanH) [reply .eslintrc:7]: It does, that's true. We shouldn't overwrite current rules

File: views/PokemonsView/PokemonsView.jsx [hunk 2]
```diff
@@ -35,6 +38,10 @@ function PokemonsView(props) {
     }
   }, []);
```
Comment (radovix) [inline views/PokemonsView/PokemonsView.jsx:39]: I think we can delete this `if (pageNumber === 0)` condition because that should always be true, right?

Comment (DorijanH) [reply views/PokemonsView/PokemonsView.jsx:39]: Not really. For example, this `useEffect` is triggered when navigating back from the Pokemon details page. In that case, as the current `pageNumber` is not equal to `0`, we don't trigger additional `loadPokemons` as they are already "cached" inside the `PokemonsViewModel`.

File: components/LoadingContainer/LoadingContainer.stories.jsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        ))}
+      </div>
+    </LoadingContainer>
+  );
+}
+
+// Not loading story
+export const NotLoading = Template.bind({});
+NotLoading.args = {
+  ...Template.args,
```
Comment (radovix) [inline components/LoadingContainer/LoadingContainer.stories.jsx:62]: No need to do object destruction here. I suggest deleting it from this and all other stories.

**Review by radovix — APPROVED** [review]

**Pull request merged**